### ISSUE TITLE
ES6 const definitions result in "..is defined but never used   no-unused-vars" error

### ISSUE THREAD

**Issue by infolock** [yannickcr/eslint-plugin-react#101]
Please do not confuse this with issues #18 and #19 that have already been reported.

When requiring in a library via `const` definitions, eslint-plugin-react will throw the following kind of error:

>   7:6  error  $ is defined but never used                     no-unused-vars

The code looks like this:
```js
const React = require('react');
const $ = require('jquery');
//....
```

This will fail with the error reported above when attemping to run lint.  However, *this* will pass:

```js
const React = require('react');
var $ = require('jquery');
//....
```

Simply changing this to `var` fixes the issue.  So I guess the bug is that `const` doesn't seem to either be acknowledged/supported?  

Edit:  Note that issue #31 seems to have been mistakenly assumed to be a duplicate of the #18 and #19 bugs - but the main thing is that he is using `const` - which is what the actual issue is.

**Comment by yannickcr** [comment]
I don't really see the relation with eslint-plugin-react here, are you sure it is not an eslint issue ?

Also, I was not be able to reproduce the problem, can you post more information ? Like the eslint/eslint-plugin-react versions used and your `.eslintrc` config file. Thanks.

**Comment by infolock** [comment]
Thanks for the reply!  You may be right about it not being an estlint-plugin-react issue - but I find it is best to report the issue with the base component itself first (estlint-plugin-react) and then work your way backwards.  It just makes it easier to pin-point exactly where the problem is.  

Either way, here is the information you requested:

```
$ npm list | grep -i eslint
|- eslint@0.20.0
|- eslint-plugin-react@2.5.0
|- grunt-eslint@12.0.0
```

**estlintConfig**
```
{
    "globals": {
        "require": true,
        "window": true,
        "module": true
    },
    "rules": {
        "quotes": 0,
        "strict": 0,
        "no-underscore-dangle": 0,
        "new-cap": 0,
        "no-mixed-spaces-and-tabs": 0,
        "no-multi-spaces": 0,
        "react/display-name": 0,
        "react/jsx-boolean-value": 1,
        "react/jsx-quotes": 1,
        "react/jsx-no-undef": 1,
        "react/jsx-sort-props": 0,
        "react/jsx-sort-prop-types": 0,
        "react/jsx-uses-react": 1,
        "react/jsx-uses-vars": 1,
        "react/no-did-mount-set-state": 1,
        "react/no-did-update-set-state": 1,
        "react/no-multi-comp": 1,
        "react/no-unknown-property": 1,
        "react/prop-types": 1,
        "react/react-in-jsx-scope": 1,
        "react/self-closing-comp": 1,
        "react/wrap-multilines": 1
    },
    "plugins": [ "react" ],
    "ecmaFeatures": {
        "jsx": true,
        "arrowFunctions": true,
        "blockBindings": true,
        "classes": true,
        "templateStrings": true,
        "objectLiteralShorthandProperties": true
    }
}
```

**Comment by infolock** [comment]
fyi, https://github.com/eslint/eslint/issues/2780  << fixed by upgrading to 0.23.0 of eslint
